### ISSUE TITLE
add solidity coverage

### ISSUE THREAD

**Issue by perissology** [Giveth/giveth-bridge#4]
look into getting https://github.com/0xProject/0x-monorepo/tree/development/packages/sol-cov/ to work.

It appears that we just need to produce a artifacts file which would be fairly trivial to generate from the solcpiler output.

artifacts content is found https://github.com/0xProject/0x-monorepo/blob/development/packages/sol-cov/src/collect_contract_data.ts#L13

**Comment by perissology** [comment]
okay, I took a shot at this w/ no luck.

I ran into a few errors when trying to write the coverage report.

1. throwing error, failing to find a NEW_CONTRACT. Contract may not have been in the artifacts file. I commented out the place where error was thrown and the next error occurred.
2. ParserError: looks like this occurred when trying to generate the AST in the sol-cover code. Maybe a better way about this would be to use the ast generated in standardJson, and visit that to collect the expressions.

I'm putting this on hold for the time being

**Comment by perissology** [comment]
okay, this is implemented in the `sol-cov` branch. I'm just waiting on [the PR](https://github.com/0xProject/0x-monorepo/pull/589/files/d49f2c40ae28bde3574907a72c0fdd84196e8303..ebc750d5bf95da76424da81550a88e6b74de8c36) to be merged and a new release to happen.

Then I can update the `sol-cov` branch to the latest format & merge to master

**Comment by perissology** [comment]
updates...

there were quite a few changes to the sol-cov project. I've updated solcpiler & the sol-cov branch to reflect the latest changes. Were waiting on [this PR]() to be merged and a release to happen.

In the meantime, to run the sol-cov branch do the following:

1. clone https://github.com/perissology/0x-monorepo/ locally
2. `cd 0x-monorepo`
3. `git co feature/missing-sources`
4. follow direction in [README.md](https://github.com/perissology/0x-monorepo/tree/feature/missing-sources/packages/sol-cov) to install & build
5. `yarn link`
6 `cd ../path/to/giveth-bridge`
7. `yarn link "@0xproject/sol-cov"

You should now be able to run `yarn cover:sol`
